core/src/HPX/Kokkos_HPX.hpp

```diff
@@ -1284,7 +1296,9 @@ class ParallelScan<FunctorType, Kokkos::RangePolicy<Traits...>,
       // Since arrive_and_wait may yield and resume on another worker thread we
       // set in_parallel = false on the current thread before suspending and set
       // it again to true when we resume.
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
       Kokkos::Experimental::HPX::impl_not_in_parallel_scope p;
+#endif
       barrier.arrive_and_wait();
     }
```

Comment:
No please guard around the whole block and do `barrier.arrive_and_wait();` in the `#else` branch.

For reference this is the PR that added support for `HPX::in_parallel()`
https://github.com/kokkos/kokkos/pull/6143/files#diff-76a7214a6cee6d4ac5794386cf6a2e6cd335022e3bf6b32275cf0106ce367de1L1262-R1301

Reply:
Please include the comment in the deprecated branch in all these cases (and preferably have the curly braces only in the deprecated branch).